src/main/java/com/cmlteam/serv/HttpHandlerFilesList.java

```diff
@@ -0,0 +1,121 @@
+package com.cmlteam.serv;
+
+import com.sun.net.httpserver.HttpExchange;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+public class HttpHandlerFilesList extends HttpHandlerBase {
+
+    private final File[] files;
+    private Deque<String> referers = new ArrayDeque<>();
```

Comment:
This is incorrect. Handler is global thing which means it's simultaneously accessed by all clients. This means that **referers** of different clients will clash, which means that when hitting back user can see incorrect page (of other user).
You can easily emulate this even by one user just by opening the web page in 2 tabs and playing with them.

Handler should be stateless. You don't need to remember state on server instead the back link should be the regular link to top-level folder like: `/listing?name=/path/to/some/folder`